requirements.txt

```diff
@@ -1,8 +1,8 @@
-bittensor>=7.0.1
-starlette~=0.37.2
-pydantic~=2.7.3
-rich~=13.7.1
-pytest~=8.2.2
-torch~=2.3.1
-numpy~=1.26.4
-setuptools~=68.2.0
\ No newline at end of file
+bittensor>=7
+starlette>=0.30.0
+pydantic>=2
+rich>=13
+pytest>=8
+torch>=2
+numpy>=1
```

Comment:
I am not sure about this `numpy` version. See this https://github.com/opentensor/bittensor/pull/2045/files Community had problems with numpy version 2.

Reply:
This is not for the bittensor repo. This is the template repo. Here anything greater than v1 will be fine.

Reply:
I realize that. My worry (maybe not valid) is that a common use case for this template repo is for a user to install bittensor, create wallet and then use this template repo. If this template repo reinstalls numpy version 2, then any "old subnet code" they have that uses numpy will likely break.

Reply:
bittensor has a hard requirement for numpy 1.26 and this template repo has a requirement of bittensor. 

when the package resolution occurs itll find the version that works for all packages/dependencies and it will see that because bittensor requires 1.26, and the template repo requires >=1, it will then install 1.26 to be compliant with all dependencies.

Reply:
if it cannot install the correct version, it will throw an error stating that a dependency has a conflict and cannot be installed. 

The reason I loosened these requirements is because we ran into such an issue.

Reply:
numpy v1.26 >= numpy v1